command/agent/fs_endpoint.go

```diff
@@ -17,12 +17,12 @@ import (
 )
 
 var (
-	allocIDNotPresentErr  = fmt.Errorf("must provide a valid alloc id")
-	fileNameNotPresentErr = fmt.Errorf("must provide a file name")
-	taskNotPresentErr     = fmt.Errorf("must provide task name")
-	logTypeNotPresentErr  = fmt.Errorf("must provide log type (stdout/stderr)")
-	clientNotRunning      = fmt.Errorf("node is not running a Nomad Client")
-	invalidOrigin         = fmt.Errorf("origin must be start or end")
+	allocIDNotPresentErr  = CodedError(400, "must provide a valid alloc id")
```

Comment:
Should these be 412? Precondition Failed

Reply:
412 (Precondition Failed) seems applicable to conditional requests i.e. ones with `If-Unmodified-Since` [1][2][3].

[1] https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412
[2] https://stackoverflow.com/questions/5369480/when-is-it-appropriate-to-respond-with-a-http-412-error
[3]https://tools.ietf.org/html/rfc7232#section-4.2